### DISCUSSION TITLE
为什么要去掉 Dropdown 控件的 overlay 属性， 而使用 menu 呢？

### DISCUSSION THREAD

**Discussion by ekuku1984** [ant-design/ant-design#40485] [General]
为什么要去掉 Dropdown 控件的 overlay 属性， 而使用 menu 呢？
使用 Dropdown 控件并不只是下拉菜单， 而且会下拉很多其他的内容， 文档里说可以使用 Popover 组件替换， 但是感觉 Popover 并不能替代 Dropdown， Popover 弹出的效果，并不是下拉、arrow 不能去掉等，感觉 Popover 是替换不了 Dropdown。

项目项目里使用了 Dropdown 很多并不是 menu, 为了不出现警告， 使用了 dropdownRender 这个属性来返回 Dropdown 的内容，menu 设置为 {{ items: [] }}， 目前测试也没有什么问题。

<Dropdown menu={{ items: [] }} dropdownRender={() => <Component />} >
  <Button>Test</Button>
</Dropdown>

希望后面不要再改了， 验证 dropdownRender 返回的内容只能是 menu 了。

**Comment by zombieJ** [comment]
设计上来看 Dropdown 目的是为了做下拉菜单，但是在最早实现中因为想复用 Menu 的样式所以直接用了 `overlay` 传入 ReactNode 的实现。这就使得用户会误用该能力实现非下拉菜单的效果。你提到的这点正是要给 Dropdown 收口的原因。

在 `4.24.0` 版本中，`overlay` 收口为 `menu`，同时提供 `dropdownRender` 做自定义渲染需求（也包含非标的需求）。`dropdownRender` 本身不会校验元素内容，请放心。
